packages/runtime-core/src/scheduler.ts

```diff
@@ -162,7 +162,7 @@ export function flushPreFlushCbs(
     }
     for (
       preFlushIndex = 0;
-      preFlushIndex < activePreFlushCbs.length;
+      preFlushIndex < (activePreFlushCbs && activePreFlushCbs.length);
```

Comment:
activePreFlushCbs = [...new Set(pendingPreFlushCbs)]

it just got assign before this, what could cause it to be null here?

Reply:
I know. I've seen that. But it is `null` in my application. I suspect it may be caused by [AG Grid](https://github.com/ag-grid/ag-grid) library and the way it internally handles Vue components which are used as table cells.

Reply:
When it should not be possible for a variable to be null, but it is, I don't think that making it not break when it's null is a proper solution. 

That would likely mask the source of the bug and make it even harder to track down when it reappears later in a different variation.

Reply:
and yet I have found the value to be null at times, and I do not use AG Grid.

Reply:
I have tested another way to fix:

`.       let length = activePreFlushCbs.length;
          for (preFlushIndex = 0; preFlushIndex < length; preFlushIndex++) {
              if (checkRecursiveUpdates(seen, activePreFlushCbs[preFlushIndex])) {
                  continue;
              }
              activePreFlushCbs[preFlushIndex]();
          }` 

activePreFlushCbs is ALWAYS an array just before the loop starts, as expected. Somehow, activePreFlushCbs is becoming null inside the loop.

Reply:
In a project that reproduces the issue, add console.log before and after this line inside the for loop:
`console.log('BEFORE');
activePreFlushCbs[preFlushIndex]();
console.log('AFTER');`
One would expect the console to contain repeating sequences of BEFORE and AFTER. However, when the error happens it produces a sequence of: BEFORE BEFORE AFTER, then error.
A call to flushPreFlushCbs is being made before a previous call is finishing. That is why activePreFlushCbs can sometimes be null. If this is by design, then the fix introduced by either livthomas or dsonet will work fine.

Reply:
Our team is seeing this issue as well. We do use AgGrid, but I'm not sure that is related. I see `activePreFlushCbs` is declared on line 43 outside the functions using it. `flushPreFlushCbs` is recursive and a couple other functions call it in the same file. `flushPreFlushCbs` is also the function that sets `activePreFlushCbs` to `null`. `queuePreFlushCb` also looks like, when its called, it eventually ends up running `flushPreFlushCbs`. It seems @rhues is likely correct and `flushPreFlushCbs` may be pulling the rug out from under itself.

Reply:
In case its helpful, a teammate seems to have isolated what causes the issue to appear, calling AgGrid's `gridApi.flushAsyncTransactions()` It only happens in a specific case and `flushAsyncTransactions()` is called many other times with out the `null` issue appearing, but in the case where the issue appears, commenting out the call to `flushAsyncTransactions()` seems to avoid the issue.